Program.cs

```diff
@@ -1,4 +1,145 @@
+// ------------------------------------------------------------------------
+// Program.cs
+// MYLIST<T> 
+// Implement a custom MyList<T> class using arrays as the underlying data structure.
+// --------------------------------------------------------------------------------------------
+using System;
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   class Program {
+      #region Method ---------------------------------------------
+      static void Main () {
+         MyList<int> list = new ();
+         for (int i = 1; i <= 7; i++) list.Add (i);
+         list.Display ();
+         list.Remove (7);
+         list.Clear ();
+         list.Display ();
+         for (int i = 1; i <= 6; i++) list.Insert (i - 1, i);
+         list.Display ();
+         list.RemoveAt (0);
+         list.RemoveAt (1);
+         list.Display ();
+      }
+      #endregion
+   }
+   #endregion
 
-   
\ No newline at end of file
+   #region MyList<T> ------------------------------------------------------------------------------
+   /// <summary>Create a class</summary>
+   /// <typeparam name="T">Datatype of the list</typeparam>
+   public class MyList<T> {
+      #region Constructor -------------------------------------------
+      /// <summary>Creating a Constructor for initializing the values</summary>
+      public MyList () {
+         mArray = new T[4];  // Initialize the array capacity as 4. 
+         mCount = 0;
+      }
+      #endregion
+
+      #region Properties --------------------------------------------
+      /// <summary>Calculating capacity of the array</summary>
+      public int Capacity => mArray.Length;
+
+      /// <summary>Gets the number of elements in the list</summary>
+      public int Count => mCount;
+
+      /// <summary>Set and return the value based on the index</summary>
+      /// <param name="index">Index to be change</param>
+      /// <returns>Changed value of the index</returns>
+      public T this[int index] {
+         get {
+            CheckIndexRange (index);
+            return mArray[index];
+         }
+         set {
+            CheckIndexRange (index);
+            mArray[index] = value;
+         }
+      }
+      #endregion
+
+      #region Implementation ----------------------------------------
+      /// <summary>Add the value to the end</summary>
+      /// <param name="a">Value to be add</param>
+      public void Add (T a) {
+         CheckCapacity (mCount + 1);
+         mArray[mCount] = a;
+         mCount++;
+      }
+
+      /// <summary>Check the array capacity</summary>
+      /// <param name="minCapacity">The minimum capacity required</param>
+      private void CheckCapacity (int minCapacity) {
+         if (minCapacity > mArray.Length) {
+            int newCapacity = mArray.Length * 2;  // Double the capacity when needed
+            if (newCapacity < minCapacity) newCapacity = minCapacity;

```

Comment:
I don't understand why this check is done. Will `newCapacity` ever be less than `minCapacity`?